### PR TITLE
updating mathcomp to opam 2 and coq 8.9

### PR DESCRIPTION
**Pull request by CohenCyril**

### CODE REVIEW

File: released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.7.0/opam [hunk 1]
```diff
@@ -2,16 +2,17 @@ opam-version: "1.2"
...
 dev-repo: "https://github.com/math-comp/math-comp.git"
 license: "CeCILL-B"
 
-build: [ make "-C" "mathcomp/ssreflect" "-j" "%{jobs}%" ]
-install: [ make "-C" "mathcomp/ssreflect" "install" ]
+build: [ make "-C" "mathcomp/ssreflect" "-C" "mathcomp/ssreflect" "-j" "%{jobs}%" ]
+install: [ make "-C" "mathcomp/ssreflect" "-C" "mathcomp/ssreflect" "install" ]
 remove: [ "sh" "-c" "rm -rf '%{lib}%/coq/user-contrib/mathcomp'" ]
-depends: [ "coq" { ((>= "8.6" & < "8.9~") | (= "dev"))} ]
+depends: [ "coq" { ((>= "8.6" & < "8.10~") | (= "dev"))} ]
```
Comment (ejgallego) [inline released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.7.0/opam]: I am not sure I understand why we put an upper version limit here [thus requiring updating when a new Coq is released]

File: released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.7.0/opam [hunk 2]
```diff
@@ -2,16 +2,17 @@ opam-version: "1.2"
...
-homepage: "http://math-comp.github.io/math-comp/"
+homepage: "https://math-comp.github.io/math-comp/"
 bug-reports: "Mathematical Components <mathcomp-dev@sympa.inria.fr>"
 dev-repo: "https://github.com/math-comp/math-comp.git"
 license: "CeCILL-B"
 
-build: [ make "-C" "mathcomp/ssreflect" "-j" "%{jobs}%" ]
-install: [ make "-C" "mathcomp/ssreflect" "install" ]
+build: [ make "-C" "mathcomp/ssreflect" "-C" "mathcomp/ssreflect" "-j" "%{jobs}%" ]
+install: [ make "-C" "mathcomp/ssreflect" "-C" "mathcomp/ssreflect" "install" ]
```
Comment (ejgallego) [inline released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.7.0/opam]: That double `-C` looks weird, doesn't it?

Comment (CohenCyril) [reply released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.7.0/opam]: right... this is automatically generated, I should doublecheck the generator :smile:

File: released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.7.0/opam (hunk 1 above)
Comment (CohenCyril) [reply released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.7.0/opam]: I do not know what best practices are... should I remove the upper bound? There is not guarantee a given version of mathcomp will be compatible with future versions of Coq, though... When a new Coq is released and tested against older mathcomp releases, we may bump up the upper bound...

Comment (ejgallego) [reply released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.7.0/opam]: Indeed the "upper bound" debate is old an unresolved in the packaging world as far as I can tell.

I think indeed not having upper bounds by default is a common trend in package universes where compatibility "is desired"; adding such constraints only in the case of breakage. This works well for example in C libraries which tend to be ABI-compatible.

**Comment by gares** [comment]
I did support the upper bound business, ant put it in the packaging doc, because at that time coq releases were infrequent and breaking code.

The situation has changed, so I'm happy to rediscuss it and weaken this requirement. Maybe a small slot in the next WG.

**Comment by ejgallego** [comment]
> The situation has changed, so I'm happy to rediscuss it and weaken this requirement. Maybe a small slot in the next WG.

Sounds good; to be clear, I am OK with the upper bound, just wondering if that's indeed the best choice today.

**Pull request merged**